tests/hash_to_curve_g1.rs

```diff
@@ -1,56 +1,78 @@
-use bls12_381::G1Projective;
+use bls12_381::{
+    hash_to_curve::{ExpandMsgXmd, HashToCurve},
+    G1Affine, G1Projective,
+};
+use hex_literal::hex;
+use sha2::Sha256;
 
-// test vectors from the draft 10 RFC
-#[test]
-fn test_encode_to_curve_10() {
-    use bls12_381::{
-        hash_to_curve::{ExpandMsgXmd, HashToCurve},
-        G1Affine,
-    };
-    use std::string::{String, ToString};
+struct TestCase {
+    msg: &'static [u8],
+    dst: &'static [u8],
+    expected: &'static [u8],
+}
 
-    struct TestCase {
-        msg: &'static [u8],
-        expected: [&'static str; 2],
-    }
-    impl TestCase {
-        fn expected(&self) -> String {
-            self.expected[0].to_string() + self.expected[1]
+impl TestCase {
+    pub fn check_output(&self, output: &[u8]) {
+        if output != self.expected {
+            panic!(
+                "Test vector result mismatch.\n\
+                    Message: {:x?}\n\
+                    DST: {:x?}\n\
+                    Expected: {:x?}\n\
+                    Found: {:x?}",
+                self.msg, self.dst, self.expected, output
+            )
         }
     }
+}
 
-    const DOMAIN: &[u8] = b"QUUX-V01-CS02-with-BLS12381G1_XMD:SHA-256_SSWU_NU_";
+// From <https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-16#appendix-J.9.1>
+#[test]
+fn hash_to_curve_works_for_draft16_testvectors_g1_sha256_ro() {
```

Comment:
I just realised that the `encode` and `hash` tests were swapped in order, which makes confirming that the test vectors didn't change _very_ difficult. I'm going to force-push to fix this.

The only reason I noticed this is because I factored out the `tests/` move into #134.